tests/test_utils.py

```diff
@@ -540,3 +544,113 @@ def test_get_sys_path_for_python_executable():
     # not testing for equality, because pytest adds extra paths into current sys.path
     for path in result:
         assert path in sys.path
+
+
+@pytest.mark.parametrize(
+    ("pyproject_param", "new_default"),
+    (
+        # From sync
+        ("ask", True),
+        ("dry-run", True),
+        ("find-links", ["changed"]),
+        ("extra-index-url", ["changed"]),
+        ("trusted-host", ["changed"]),
+        ("no-index", True),
+        ("python-executable", "changed"),
+        ("verbose", True),
+        ("quiet", True),
+        ("user", True),
+        ("cert", "changed"),
+        ("client-cert", "changed"),
+        ("pip-args", "changed"),
+        # From compile, unless also in sync
+        ("pre", True),
+        ("rebuild", True),
+        ("extras", ["changed"]),
+        ("all-extras", True),
+        ("index-url", "changed"),
+        ("header", False),
+        ("emit-trusted-host", False),
+        ("annotate", False),
+        ("annotation-style", "line"),
+        ("upgrade", True),
+        ("upgrade-package", ["changed"]),
+        ("output-file", "changed"),
+        ("newline", "native"),
+        ("allow-unsafe", True),
+        ("strip-extras", True),
+        ("generate-hashes", True),
+        ("reuse-hashes", False),
+        ("max-rounds", 100),
+        ("build-isolation", False),
+        ("emit-find-links", False),
+        ("cache-dir", "changed"),
+        ("resolver", "backtracking"),
+        ("emit-index-url", False),
+        ("emit-options", False),
+        ("unsafe-package", ["changed"]),
+    ),
+)
+def test_callback_config_file_defaults(pyproject_param, new_default, make_config_file):
+    config_file = make_config_file(pyproject_param, new_default)
```

Comment:
The need to inject this invocation into every test could be eliminated by making the params indirect (this needs the respective fixtures to depend on `request` and read `request.param`).

Reply:
It's a clever idea, and if I we were making the config file based on the key and value being passed in parametrized values in multiple tests, I'd agree it may be worth the effort. But this is the only invocation that gets all of its arguments from test parameters - the others either have some or all that are constants. Given that, indirect parameterization feels like over-engineering that makes the test scenario more opaque. Thoughts?

Reply:
Okay, this one not as important. Marking as resolved for now.